kingpin/actors/aws/iam/entities.py

```diff
@@ -0,0 +1,601 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2014 Nextdoor.com, Inc
+
+"""
+:mod:`kingpin.actors.aws.iam.entities`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+"""
+
+import json
+import os
+import logging
+
+from boto.exception import BotoServerError
+from tornado import concurrent
+from tornado import gen
+
+from kingpin.actors import exceptions
+from kingpin.actors.aws.iam import base
+from kingpin.constants import REQUIRED
+from kingpin.constants import STATE
+
+log = logging.getLogger(__name__)
+
+__author__ = 'Matt Wise <matt@nextdoor.com>'
+
+
+# This executor is used by the tornado.concurrent.run_on_executor()
+# decorator. We would like this to be a class variable so its shared
+# across RightScale objects, but we see testing IO errors when we
+# do this.
+EXECUTOR = concurrent.futures.ThreadPoolExecutor(10)
+
+
+class EntityBaseActor(base.IAMBaseActor):
```

Comment:
There's a ton of try/catch-boto cases here. I think it could be simplified if you had a wrapper around `self.thread` something like `self.thread_boto` which would try/catch boto specific issues, and maybe handle retry on common cases there.